drivers/src/sha512.rs

```diff
@@ -0,0 +1,40 @@
+// Licensed under the Apache-2.0 license
+
+use caliptra_error::{CaliptraError, CaliptraResult};
+use caliptra_registers::sha512::Sha512Reg;
+
+use crate::{Array4x12, Array4x8};
+
+pub struct Sha512 {
+    sha512: Sha512Reg,
+}
+
+impl Sha512 {
+    pub fn new(sha512: Sha512Reg) -> Self {
+        Self { sha512 }
+    }
+
+    pub fn gen_pcr_hash(&mut self, nonce: Array4x8) -> CaliptraResult<Array4x12> {
```

Comment:
I think this should just be moved to the sha384 driver (which also uses the sha512 peripheral) since it's generating a sha384 hash.